server/src/lib/model/user-client.ts

```diff
@@ -243,6 +245,18 @@ const UserClient = {
     return true;
   },
 
+  // update the challenge_team of the user who is already signed up but have not joined any team
+  // [BUG]: there are exceptions if the team doesn't exist
+  async updateChallengeTeam(
+    email: string,
+    challenge_team: string
+  ): Promise<boolean> {
+    return await db.query(
+      'UPDATE user_clients SET challenge_team = ? WHERE email = ? AND has_login AND challenge_team IS NULL',
```

Comment:
Should this be `… AND challenge_team IS NOT NULL`?

Reply:
My understanding is that NULL challenge_team means the user has not enrolled yet, he/she may have signed up independently, not from the invitation link.

This also implies that users can't switch teams, enrollment happens only once.
I forgot to ask you about the implications, sorry.